receiver/azuremonitorreceiver/scraper.go

```diff
@@ -98,14 +99,31 @@ func (*timeWrapper) Now() time.Time {
 	return time.Now()
 }
 
+type storageAccountHackConfig struct {
```

Comment:
I feel like we should we add a comment around this considering we are outwardly calling it a hack.